### DISCUSSION TITLE
Failed to start gauge API: Timed out connecting to java

### DISCUSSION THREAD

**Discussion by ghost** [getgauge/gauge#1835] [General]
While I'm running
`gradle --offline -q -PgaugeTags=tag -PgaugeEnv=default gaugeTask`

I'm getting the following error:
Error ----------------------------------
[Gauge]
Failed to start gauge API: Timed out connecting to java
Get Support ----------------------------
        Docs:          https://docs.gauge.org
        Bugs:          https://github.com/getgauge/gauge/issues
        Chat:          https://spectrum.chat/gauge

Your Environment Information -----------
        linux, 1.1.6, 3af456f
        html-report (4.0.12), java (0.7.13), screenshot (0.0.1)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gaugeTask'.
> Execution failed for one or more tests!

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 38s

Playing with the gauge.properties timeout values does not help for me.
The log from the gauge.log -> 04-01-2021 13:18:24.217 [Gauge] [CRITICAL] Failed to start gauge API: Timed out connecting to java

**Comment by sriv** [comment]
What version of `gauge-java` reference does your `build.gradle` have @Borisakov ? Can you please ensure that it is the latest and try?

**Comment by ghost** [comment]
@sriv  Yes, it is the latest
"com.thoughtworks.gauge:gauge-java:0.7.13"

**Reply by sriv** [reply]
Do you see any `.out` files in the `logs` directory? If you do, can you please check if it has any errors logged?

**Reply by ghost** [reply]
@sriv
This is the only file I have: (I removed the spec names)
-rw-r--r-- 1 root root 6086 Jan 4 14:42 gauge.log
root@3dbaaac9b834:/automation-system-tests/build/logs# cat gauge.log
04-01-2021 14:39:33.219 [Gauge] [DEBUG] Created gauge_screenshots_dir at /automation-system-tests/build/screenshots
04-01-2021 14:39:33.220 [Gauge] [DEBUG] Plugin java is already installed.
04-01-2021 14:39:33.220 [Gauge] [DEBUG] Plugin html-report is already installed.
04-01-2021 14:39:33.220 [Gauge] [DEBUG] Plugin screenshot is already installed.
04-01-2021 14:39:33.220 [Gauge] [DEBUG] Parsing started.
04-01-2021 14:39:33.220 [Gauge] [DEBUG] Started concepts parsing.
04-01-2021 14:39:33.437 [Gauge] [DEBUG] 597 concepts parsing completed.
04-01-2021 14:39:33.438 [Gauge] [DEBUG] Started specifications parsing.
04-01-2021 14:39:33.457 [Gauge] [DEBUG] 90 specifications parsing completed.
04-01-2021 14:39:33.457 [Gauge] [DEBUG] Applying tags filter: ***
04-01-2021 14:39:33.472 [Gauge] [DEBUG] The following specifications satisfy filter criteria:
04-01-2021 14:39:33.472 [Gauge] [DEBUG] specs//.spec
04-01-2021 14:39:33.472 [Gauge] [DEBUG] specs//.spec
04-01-2021 14:39:33.472 [Gauge] [DEBUG] specs//.spec
04-01-2021 14:39:33.472 [Gauge] [DEBUG] specs//.spec
04-01-2021 14:39:33.472 [Gauge] [DEBUG] Parsing completed.
04-01-2021 14:39:33.476 [java] [DEBUG] ***** Printing Properties/Env Values *****
04-01-2021 14:39:33.476 [java] [DEBUG] PATH: /.gauge:/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/.gauge/plugins/screenshot/0.0.1
04-01-2021 14:39:33.476 [java] [DEBUG] JAVA_HOME: /opt/java/openjdk
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_project_root:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_parallel_streams_count:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_reports_dir: build/reports
04-01-2021 14:39:33.476 [java] [DEBUG] overwrite_reports: true
04-01-2021 14:39:33.476 [java] [DEBUG] logs_directory: build/logs
04-01-2021 14:39:33.476 [java] [DEBUG] enable_multithreading: false
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_specs_dir: specs
04-01-2021 14:39:33.476 [java] [DEBUG] csv_delimiter: ,
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_java_home:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_custom_build_path:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_additional_libs: libs/*
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_jvm_args:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_custom_compile_dir:
04-01-2021 14:39:33.476 [java] [DEBUG] gauge_clear_state_level:
04-01-2021 14:39:33.476 [java] [DEBUG] ***** END *****
04-01-2021 14:39:33.684 [Gauge] [DEBUG] Checking updates...
04-01-2021 14:39:33.684 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/html-report?l=java&p=html-report,java,screenshot&o=linux&a=amd64
04-01-2021 14:39:34.223 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/java?l=java&p=html-report,java,screenshot&o=linux&a=amd64
04-01-2021 14:39:34.372 [Gauge] [DEBUG] Downloading https://downloads.gauge.org/plugin/screenshot?l=java&p=html-report,java,screenshot&o=linux&a=amd64
04-01-2021 14:40:03.476 [Gauge] [CRITICAL] Failed to start gauge API: Timed out connecting to java

**Reply by sriv** [reply]
This is a bit strange, what is the value of `runner_connection_timeout` in the machine? It is set to 30 seconds by default, which is the interval I see it is timing out here. My guess is that on that machine 30 second is perhaps not enough for the java process to compile the test project and bring up the runner, and gauge is timing out prematurely.

Can you try setting `gauge config runner_connection_timeout <val in ms>` in the machine? This can also be set by editing `~/.gauge/config/gauge.properties` file. 

Note that this setting is at a machine level.

**Reply by ghost** [reply]
Thanks @sriv 
There is a gauge.properties file in the root of the project with the following value
runner_connection_timeout = 200000
But when I run the command `gauge config --list` it prints
runner_connection_timeout     	30000
What am I missing?

**Reply by sriv** [reply]
`gauge.properties` is not read from project root. Gauge looks for it in `$HOME/.gauge/config`
